src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -80,19 +85,38 @@ SyntaxNodeAnalysisContext ctx
 				return;
 			}
 
+			bool requireNamedArguments = HasRequireNamedArgumentsAttribute( ctx.SemanticModel, args );
```

Comment:
This would be more efficient after the unnamedArgs calculation (this code runs for every fn/etc. call) but only if we stick with the pseudo naming logic (right now this is being passed into that fn so can’t be moved. )

The intent of the pseudo naming logic is that as a reader (that knows about this behaviour) there’s no ambiguity... adding names is just noise. 

It’s more important when we apply this widely i.e. not just for methods with an attribute because it cuts down on noise (think adapters/etc. ) so I don’t feel super strongly in this case.